### ISSUE TITLE
MUP ignoring MONGO_URL settings on deployment

### ISSUE THREAD

**Issue by shankscoder** [zodern/meteor-up#849]
I'm trying to deploy a Meteor 1.6.0.1 app using MUP. I'm getting an error where the deployment fails as its trying to connect to the wrong Mongo endpoint.

I'm looking to self-host Meteor and the Mongo setup myself. I have 2 servers on Vultr one for the webserver, and one running Mongo. The Mongo server is accessible from the webserver and I'm able to use the connection details from external clients.

However, when I try to deploy using MUP, I get the following error:

```
x Verifying Deployment: FAILED
	
	-----------------------------------STDERR-----------------------------------
	ting meteor app on port:80
	
	/built_app/programs/server/node_modules/fibers/future.js:313
							throw(ex);
							^
	MongoError: failed to connect to server [localhost:27017] on first connect [MongoError: connect ECONNREFUSED 127.0.0.1:27017]
	    at Pool.<anonymous> (/built_app/programs/server/npm/node_modules/meteor/npm-mongo/node_modules/mongodb-core/lib/topologies/server.js:336:35)
	    at emitOne (events.js:115:13)
	    at Pool.emit (events.js:210:7)
	    at Connection.<anonymous> (/built_app/programs/server/npm/node_modules/meteor/npm-mongo/node_modules/mongodb-core/lib/connection/pool.js:280:12)
	    at Object.onceWrapper (events.js:318:30)
	    at emitTwo (events.js:125:13)
	    at Connection.emit (events.js:213:7)
	    at Socket.<anonymous> (/built_app/programs/server/npm/node_modules/meteor/npm-mongo/node_modules/mongodb-core/lib/connection/connection.js:187:49)
	    at Object.onceWrapper (events.js:316:30)
	    at emitOne (events.js:115:13)
	 
	=> Redeploying previous version of the app
	 
	-----------------------------------STDOUT-----------------------------------
	
	To see more logs type 'mup logs --tail=100'
	
	----------------------------------------------------------------------------
```

Here's my mup.js file (private data obfuscated). The file resides in a `.deploy` folder in my Meteor app root:

```
module.exports = {
  servers: {
    one: {
      host: '<SERVERIP>',
      username: '<USERNAME>',
      pem: '<PATH>'
    }
  },
  meteor: {
    name: '<APPNAME>',
    path: '../',
    docker: {
      image: 'abernix/meteord:node-8-base',
    },
    servers: {
      one: {}
    },
    buildOptions: {
      serverOnly: true
    },
    env: {
      PORT: 3000,
      MONGO_URL: 'mongodb://<DBUSER>:<DBPASSWORD>@<DBIP>:27017/<DBNAME>?replicaSet=<REPLICANAME>',
      MONGO_OPLOG_URL:  'mongodb://<OPLOGUSER>:<OPLOGPASS>@<DBIP>:27017/local?authSource=admin&replicaSet=<REPLICANAME>',
      ROOT_URL: 'xxxxx'
    },
    deployCheckWaitTime: 60
  }
}
```

I'm a bit puzzled as I've clearly mentioned the correct connection URI in the environment variables, and removed the reference to the local mongo setup. I'm not clear why the build script still insists on connecting to localhost not the remote Mongo URL.

What am I missing here?

**Comment by shankscoder** [comment]
So, I managed to get this resolved, but I had to move away from MUP and use PM2-Meteor (https://www.npmjs.com/package/pm2-meteor) to get this to work.

I suspect the issue had to do with my Oplog setup (I rebuilt the user and got this working as well), but I was able to get it to work with PM2. An interesting secondary issue was that the Mongo connections started working when I moved my external DB reference from its IP to a DNS FQDN. I have no idea if that's what fixed it, but I was able to authenticate and connect to the DB the moment after I changed to a FQDN.

I'll check again on another installer and see if the issue is with the Oplog setup or not.

I'll close this for now, and re-open if it replicates on a fresh application as well.
